pkgs/by-name/co/codecrafters-cli/package.nix

```
@@ -0,0 +1,38 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, git
+}:
+
+buildGoModule rec {
+  pname = "codecrafters-cli";
+  version = "34";
+  commit = "38dc748";
+
+  src = fetchFromGitHub {
+    owner = "codecrafters-io";
+    repo = "cli";
+    rev = "v${version}";
+    sha256 = "sha256-rNWGd+yUDIrL2P+1XEce+HH8KDfThrTPXW+EOrV/n3o=";
```

`hash` is preferred over `sha256` as it provides more flexibility to change the hashing algorithm in the future
```suggestion
    hash = "sha256-rNWGd+yUDIrL2P+1XEce+HH8KDfThrTPXW+EOrV/n3o=";
```